internal/util/logging/logram.go

```diff
@@ -0,0 +1,69 @@
+// Copyright 2021 FerretDB Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// LogRAM implements zap logging entry interception and stores the last 1024 entries in the ring buffer in memory.
+package logging
+
+import (
+	"fmt"
+	"sync"
+
+	"go.uber.org/zap/zapcore"
+)
+
+var LogRAM *logRAM
```

Comment:
I miss information about architectural/structuring choices and the reasons behind them. Maybe, it's possible to document your choices in the PR description?

For example, why is this variable defined here? Doesn't the main package just call the creation of an instance of `logRAM` with `NewLogRAM`? And then the main package is fully responsible for it.

Here you define a variable but some other package is responsible to initialize it. It doesn't look really consistent to me, and the lifecycle of this variable is not clear.

Reply:
> Re-checked everything once again. I think, my main question is about the `var LogRAM *logRAM` defined in this package. I would like to see some explanation of this singleton tech choice.

LogRam is the last 1024 stored in memory.
Declared as a global variable, it was possible to access the getLog command and get these messages using the Poue method.

Reply:
> > Re-checked everything once again. I think, my main question is about the `var LogRAM *logRAM` defined in this package. I would like to see some explanation of this singleton tech choice.
> 
> LogRam is the last 1024 stored in memory. Declared as a global variable, it was possible to access the getLog command and get these messages using the Poue method.

I am ready to implement differently if another way is offered

Reply:
Well, for example, in the `logging` package we can define a structure that would contain logger's properties including the storage for the hook you implemented. 

Something like this (maybe, with a better naming):

```go
struct Logging {
    l *zap.Logger
    circularBuffer *logRAM
}

func New() *Logging {
    return &Logging{
        // necessary params, hook settings etc
    }
}

func (l *Logging) Logger() *zap.Logger {
    return l.l
}
```

Then the caller (the `main` func) will do something like
```go
   logging := logging.New()
   logger := logging.Logger()
```

And then this logger is the one to use across the whole app.
With that, there is no package that can directly access and affect `circularBuffer`.

But I'm not sure if this was the initial idea. @AlekSi could you have a look?

Reply:
`logging.Setup` initializes the global logger and, by extension, the whole tree of derived zap loggers. It is likely we will change that later once we need more control over them; but for that PR, let's leave `Setup` to initialize a package-level state, including `LogRAM`.